### PR TITLE
RIA-3510: Handle null values in HO response

### PR DESCRIPTION
**Pull request by divisathyan**
Handle null values in HO response

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-3510

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/AsylumCaseStatusSearchHandler.java [hunk 1]
```diff
@@ -70,31 +71,38 @@ public boolean canHandle(
...
-                    HomeOfficeDateFormatter.getIacDateTime(applicationStatus.getDecisionDate()));
-                selectedMainApplicant.setDisplayDecisionSentDate(
-                    HomeOfficeDateFormatter.getIacDateTime(applicationStatus.getDecisionCommunication().getSentDate())
-                );
-
-                Optional<HomeOfficeMetadata> metadata = selectMetadata(applicationStatus.getHomeOfficeMetadata());
-                if (metadata.isPresent()) {
-                    selectedMainApplicant.setDisplayMetadataValueBoolean(
-                        ("true".equals(metadata.get().getValueBoolean())) ? "Yes" : "No"
+                if (isNull(person) || isNull(applicationStatus)) {
```
Comment (jkumar-moj) [inline src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/AsylumCaseStatusSearchHandler.java:74]: @divisathyan This is good as a generic message. But could we not check the error section of the response for any error code and corresponding message and derive a meaningful message to user based on that. I think that would be more accurate of the issue. What do you think?

Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/AsylumCaseStatusSearchHandler.java:74]: Error message change will be part of RIA-3390

**Review by jkumar-moj — APPROVED** [review]
👍

**Review by jkumar-moj — APPROVED** [review]
👍

**Review by raghera — APPROVED** [review]

**Pull request merged**